### ISSUE TITLE
spi.lsbfirst = True fails with [Errno 22] Invalid argument

### ISSUE THREAD

**Issue by Gadgetoid** [doceme/py-spidev#49]
This is related to https://github.com/doceme/py-spidev/issues/18

The above issue is what happens when an unraised exception leaks via the interpreter and is assumed to have been raised by a method from another library.

This issue is to deal with the exception itself- ie: why does `spi.lsbfirst = True` raise an Invalid Argument from `__spidev_set_mode`?:

```
static int __spidev_set_mode( int fd, __u8 mode) {
    __u8 test;
    if (ioctl(fd, SPI_IOC_WR_MODE, &mode) == -1) {
        PyErr_SetFromErrno(PyExc_IOError);
        return -1;
    }
    if (ioctl(fd, SPI_IOC_RD_MODE, &test) == -1) {
        PyErr_SetFromErrno(PyExc_IOError);
        return -1;
    }
    if (test != mode) {
        return -1;
    }
    return 0;
}
```

**Comment by Gadgetoid** [comment]
I don't believe this ioctl is supported on the Raspberry Pi, which would explain why it throws an error:

Mentioned here: https://www.raspberrypi.org/forums/viewtopic.php?f=33&t=36288

Documented as unsupported alongside `SPI_3WIRE`, `SPI_LOOP` and `SPI_READY` here: http://what-when-how.com/Tutorial/topic-334jc9v/Raspberry-Pi-Hardware-Reference-193.html

**Comment by fschrempf** [comment]
For now https://github.com/doceme/py-spidev/pull/143 intends to take care of this by adding a note to the documentation.
